app/components/user_mentee_application/reapply_cta_component.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class UserMenteeApplication::ReapplyCtaComponent < ViewComponent::Base
+  def initialize(active_cohort:, last_application:)
+    @active_cohort = active_cohort
+    @last_application = last_application
+  end
+
+  private
+
+  attr_reader :active_cohort, :last_application
+
+  def render?
+    active_cohort.present? && application_not_in_cohort?
+  end
+
+  def cta_action_text
+    last_application.present? ? 'Reapply' : 'Apply'
+  end
+
+  def application_not_in_cohort?
+    return true if last_application.blank?
+
+    application_cohort_id = last_application.user_mentee_application_cohort_id
+    application_cohort_id.blank? || (application_cohort_id != active_cohort.id)
```

Comment:
Seems like I can use `UserMenteeApplicationCohort#application_in_cohort?` here, but since it calls `#exists?` it *will* hit the db even if all the data to determine this is loaded.

Generally like to avoid db hits from components, although it might be worth noting that it wouldn't lead to N+1 in this situation.

Reply:
good shoutout and what you have here is fine.